solution/app/src/main/java/com/google/codelabs/maps/placesdemo/CurrentPlaceActivity.kt

```diff
@@ -148,17 +161,18 @@ class CurrentPlaceActivity : AppCompatActivity() {
         ) {
             // Retrieve likely places based on the device's current location
             lifecycleScope.launch {
-                try {
-                    val response = placesClient.awaitFindCurrentPlace(placeFields)
-                    responseView.text = response.prettyPrint()
-
-                    // Enable scrolling on the long list of likely places
-                    val movementMethod = ScrollingMovementMethod()
-                    responseView.movementMethod = movementMethod
-                } catch (e: Exception) {
-                    e.printStackTrace()
-                    responseView.text = e.message
+                val response = placesClient.awaitFindCurrentPlace(placeFields)
+                val place = if (!response.placeLikelihoods.isEmpty()) {
+                    response.placeLikelihoods[0].place
```

Comment:
Instead of simply choosing the first result from the likelihoods, list, let's copy the [implementation from the tutorial](https://developers.google.com/maps/documentation/android-sdk/current-place-tutorial#get-the-current-place) to present a modal to allow the user to select one of the likely places from the list, then set `place` to be that place (and therefore the marker's location and title on the map).